test/regex/replaceAndCountPerf.chpl

```diff
@@ -0,0 +1,44 @@
+use Regex;
```

Comment:
This test is named perf but doesn't have any .perfkeys or graph file.  Is that intentional?

Reply:
Yes, I am on the fence with this one.

- I don't want to add a nightly perf test for this, it is too niche.
- I wrote this to collect some anecdotal data for my implementation. I can imagine it might be used the same way down the road. In which case, it is good to have this checked in.
- Since I already have it, and it can also be used as a correctness test (not entirely sure if it tests anything different than the actual correctness test I am adding) thought I should add it.

If you think it is cluttering our testing for no good reason, I can remove it from the PR. I don't have a great argument for checking this in.

Reply:
I'm okay with leaving it in, thanks!